tests/components/rpi_power/test_binary_sensor.py

```diff
@@ -0,0 +1,73 @@
+"""Tests for rpi_power binary sensor."""
+from datetime import timedelta
+import logging
+
+from homeassistant.components.rpi_power.binary_sensor import (
+    DESCRIPTION_NORMALIZED,
+    DESCRIPTION_UNDER_VOLTAGE,
+)
+from homeassistant.components.rpi_power.const import DOMAIN
+from homeassistant.const import STATE_OFF, STATE_ON
+from homeassistant.setup import async_setup_component
+from homeassistant.util import dt as dt_util
+
+from tests.async_mock import MagicMock
+from tests.common import MockConfigEntry, async_fire_time_changed, patch
+
+ENTITY_ID = "binary_sensor.rpi_power_status"
+
+MODULE = "homeassistant.components.rpi_power.binary_sensor.new_under_voltage"
+
+
+async def _async_setup_component(hass, detected):
+    mocked_under_voltage = MagicMock()
+    type(mocked_under_voltage).get = MagicMock(return_value=detected)
+    entry = MockConfigEntry(domain=DOMAIN)
+    entry.add_to_hass(hass)
+    with patch(MODULE, return_value=mocked_under_voltage):
+        await async_setup_component(hass, DOMAIN, {DOMAIN: {}})
+        await hass.async_block_till_done()
+    return mocked_under_voltage
+
+
+async def test_new(hass, caplog):
+    """Test new entry."""
+    await _async_setup_component(hass, False)
+    state = hass.states.get(ENTITY_ID)
+    assert state.state == STATE_OFF
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6158285)